presto-main/src/main/java/io/prestosql/sql/planner/LogicalPlanner.java

```diff
@@ -182,7 +188,12 @@ public Plan plan(Analysis analysis, Stage stage)
         TypeProvider types = symbolAllocator.getTypes();
         StatsProvider statsProvider = new CachingStatsProvider(statsCalculator, session, types);
         CostProvider costProvider = new CachingCostProvider(costCalculator, statsProvider, Optional.empty(), session, types);
-        return new Plan(root, types, StatsAndCosts.create(root, statsProvider, costProvider));
+
+        StatsAndCosts statsAndCosts = StatsAndCosts.empty();
+        if (collectStatsAndCosts) {
+            statsAndCosts = StatsAndCosts.create(root, statsProvider, costProvider);
```

Comment:
move `statsProvider`, `costProvider` creation under the `if`, they are redundant otherwise